### PR TITLE
Documentation tweaks for alter and alterF

### PR DESCRIPTION
**Pull request by sjakobi**

### CODE REVIEW

File: Data/HashMap/Strict/Base.hs [hunk 1]
```diff
@@ -276,11 +281,12 @@ alter f k m =
...
 -- | /O(log n)/  The expression (@'alterF' f k map@) alters the value @x@ at
--- @k@, or absence thereof. @alterF@ can be used to insert, delete, or update
--- a value in a map.
+-- @k@, or absence thereof.
+--
+-- 'alterF' can be used to insert, delete, or update a value in a map.
 --
 -- Note: 'alterF' is a flipped version of the 'at' combinator from
--- <https://hackage.haskell.org/package/lens-4.15.4/docs/Control-Lens-At.html#v:at Control.Lens.At>.
+-- <https://hackage.haskell.org/package/lens/docs/Control-Lens-At.html#v:at Control.Lens.At>.
```
Comment (treeowl) [inline Data/HashMap/Strict/Base.hs:289]: What's the convention for inter-package links? If `at` were to move unexpectedly to a different module or package, that would invalidate a non-versioned link. OTOH, linking to some random old version of `lens` seems pretty silly!

Comment (sjakobi) [reply Data/HashMap/Strict/Base.hs:289]: I'm not aware of any convention. Frankly, I don't expect `at` to move very soon. If it moves and we then get a report regarding the broken link, that would also seem like a good thing to me.

**Comment by sjakobi** [comment]
> While you're at it: in most documentation I see, the documentation for a function doesn't link to itself. This documentation does, with `'alter'`. Is that how they all go in the module, or did someone just goof these ones up? If the latter, let's take the opportunity to ungoof them.

I'm not really at it. I originally only wanted to fix the formatting of the `alter` property.

If you care about this, feel free to make a pass yourself.

**Comment by treeowl** [comment]
Thanks.

**Pull request merged**